src/calendar-heatmap.component.js

```diff
@@ -482,7 +493,22 @@ export class CalendarHeatmap extends Component {
         return calcItemSize(d);
       })
       .attr('fill', (d) => {
-        return d.total > 0 ? color(d.total) : 'transparent';
+        let finalColor = '#ff4500';
+        if (this.props.color === 'spectral') {
```

Comment:
Refactor such that the color scale is declared outside the attr callback.

If we declare it inside the attr callback it will get redeclared as many times as the number of visual elements

Reply:
are you talking about `finalColor` sir @maneetgoyal ?

Reply:
```
const spectralColor = generateSpectralInterpolate(
            min_value,
            max_value
          );
```
etc should be calledn outside the attr callback